x-pack/legacy/plugins/uptime/server/lib/adapters/monitor_states/search/query_context.ts

```diff
@@ -4,10 +4,20 @@
  * you may not use this file except in compliance with the Elastic License.
  */
 
-import DateMath from '@elastic/datemath';
 import { APICaller } from 'kibana/server';
 import { CursorPagination } from '../adapter_types';
 import { INDEX_NAMES } from '../../../../../common/constants';
+import { parseRelativeEndDate } from '../../../helper/get_histogram_interval';
+
+export const parseRelativeDates = (
```

Comment:
This function is called `parseRelativeDates` but it also subtracts `5m` from the given value. I'd rename it `calculateTimespanBounds` or similar. Additionally, it takes `startDate` as an argument, but it never uses it. It shouldn't take arguments it doesn't use.

Reply:
i have revised this.